litestar/contrib/pydantic/pydantic_schema_plugin.py

```diff
@@ -62,73 +66,122 @@
+    # removed in v2
+    pydantic_v1.PyObject: Schema(
+        type=OpenAPIType.STRING,
+        description="dot separated path identifying a python object, e.g. 'decimal.Decimal'",
+    ),
+    # annotated in v2
+    pydantic_v1.UUID1: Schema(
+        type=OpenAPIType.STRING,
+        format=OpenAPIFormat.UUID,
+        description="UUID1 string",
+    ),
+    pydantic_v1.UUID3: Schema(
+        type=OpenAPIType.STRING,
+        format=OpenAPIFormat.UUID,
+        description="UUID3 string",
+    ),
+    pydantic_v1.UUID4: Schema(
+        type=OpenAPIType.STRING,
+        format=OpenAPIFormat.UUID,
+        description="UUID4 string",
+    ),
+    pydantic_v1.UUID5: Schema(
+        type=OpenAPIType.STRING,
+        format=OpenAPIFormat.UUID,
+        description="UUID5 string",
+    ),
+    pydantic_v1.DirectoryPath: Schema(type=OpenAPIType.STRING, format=OpenAPIFormat.URI_REFERENCE),
+    pydantic_v1.AnyUrl: Schema(type=OpenAPIType.STRING, format=OpenAPIFormat.URL),
+    pydantic_v1.AnyHttpUrl: Schema(
+        type=OpenAPIType.STRING, format=OpenAPIFormat.URL, description="must be a valid HTTP based URL"
+    ),
+    pydantic_v1.FilePath: Schema(type=OpenAPIType.STRING, format=OpenAPIFormat.URI_REFERENCE),
+    pydantic_v1.HttpUrl: Schema(
+        type=OpenAPIType.STRING,
+        format=OpenAPIFormat.URL,
+        description="must be a valid HTTP based URL",
+        max_length=2083,
+    ),
+    pydantic_v1.RedisDsn: Schema(type=OpenAPIType.STRING, format=OpenAPIFormat.URI, description="redis DSN"),
+    pydantic_v1.PostgresDsn: Schema(type=OpenAPIType.STRING, format=OpenAPIFormat.URI, description="postgres DSN"),
+    pydantic_v1.SecretBytes: Schema(type=OpenAPIType.STRING),
+    pydantic_v1.SecretStr: Schema(type=OpenAPIType.STRING),
+    pydantic_v1.StrictBool: Schema(type=OpenAPIType.BOOLEAN),
+    pydantic_v1.StrictBytes: Schema(type=OpenAPIType.STRING),
+    pydantic_v1.StrictFloat: Schema(type=OpenAPIType.NUMBER),
+    pydantic_v1.StrictInt: Schema(type=OpenAPIType.INTEGER),
+    pydantic_v1.StrictStr: Schema(type=OpenAPIType.STRING),
+    pydantic_v1.NegativeFloat: Schema(type=OpenAPIType.NUMBER, exclusive_maximum=0.0),
+    pydantic_v1.NegativeInt: Schema(type=OpenAPIType.INTEGER, exclusive_maximum=0),
+    pydantic_v1.NonNegativeInt: Schema(type=OpenAPIType.INTEGER, minimum=0),
+    pydantic_v1.NonPositiveFloat: Schema(type=OpenAPIType.NUMBER, maximum=0.0),
+    pydantic_v1.PaymentCardNumber: Schema(type=OpenAPIType.STRING, min_length=12, max_length=19),
+    pydantic_v1.PositiveFloat: Schema(type=OpenAPIType.NUMBER, exclusive_minimum=0.0),
+    pydantic_v1.PositiveInt: Schema(type=OpenAPIType.INTEGER, exclusive_minimum=0),
 }
 
-if pydantic.VERSION.startswith("1"):  # pragma: no cover
-    # pydantic v1 values only - some are removed in v2, others are Annotated[] based and require a different
-    # logic
+if pydantic_v2 is not None:
```

Comment:
Is this check correct? The previous check was for whether it's pydantic v1.

Reply:
Check is correct, the definitions are the ones for v2. The types below are false though :grimacing: